src/test/kotlin/hr/foi/pop/backend/services/UserServiceTest.kt

```diff
@@ -90,4 +92,58 @@ class UserServiceTest {
         val userWithBadUsername = templateRequestBodyForTesting.copy(username = "bad")
         assertThrows<UserCheckException> { userService.registerUser(userWithBadUsername) }
     }
+
+    @Test
+    @Transactional
+    fun givenUserWithStatusActive_onDeactivate_shouldDeactivateUser() {
+        //given
+        val deactivateUserDefinition: ActivateUserDefinitions = ActivateUserDefinitions.DEACTIVATE
+        val userIdForUserWithActiveFlag: String = "1"
+
+        //when
+        val user: User = userService.activateOrDeactivateUser(userIdForUserWithActiveFlag, deactivateUserDefinition)
+
+        //assert
+        Assertions.assertEquals(false, user.isAccepted)
+    }
+
+    @Test
+    @Transactional
+    fun givenUserWithStatusDeactivate_onActivate_shouldActivateUser() {
+        //given
+        val activateUserDefinitions: ActivateUserDefinitions = ActivateUserDefinitions.ACTIVATE
+        val userIdForUserWithDeactivateFlag: String = "4"
+
+        //when
+        val user: User = userService.activateOrDeactivateUser(userIdForUserWithDeactivateFlag, activateUserDefinitions)
+
+        //assert
+        Assertions.assertEquals(true, user.isAccepted)
+    }
+
+    @Test
+    @Transactional
+    fun givenAlreadyActivatedUser_onActivate_shouldThrowActivateUserException() {
+        //given
+        val activateUserDefinitions: ActivateUserDefinitions = ActivateUserDefinitions.ACTIVATE
+        val userIdForUserWithActivateFlag: String = "1"
+
+        //when - assert
+        Assertions.assertThrows(ActivateUserException::class.java) {
+            userService.activateOrDeactivateUser(userIdForUserWithActivateFlag, activateUserDefinitions)
+        }
```

Comment:
You can also get the thrown exception object from this assert call and then assert that it indicates the correct problem.